sky/provision/oci/instance.py

```
@@ -0,0 +1,455 @@
+                f'cluster "{cluster_name_on_cloud}" is greater than the '
+                f'number requested by the user ({config.count}). '
+                'This is likely a resource leak. '
+                'Use "sky down" to terminate the cluster.')
+
+        logger.debug(
+            f'run_instances: Found {[inst["name"] for inst in existing_instances]} '
+            'existing instances in cluster.')
+        existing_instances.sort(key=lambda x: x['name'])
+
+        stopped_instances = []
+        for existing_node in existing_instances:
+            if existing_node['status'] == 'STOPPING':
+                _wait_until_status(region, existing_node['inst_id'], 'STOPPED')
+                stopped_instances.append(existing_node)
+            elif existing_node['status'] == 'STOPPED':
+                stopped_instances.append(existing_node)
+            elif existing_node['status'] in ('PROVISIONING', 'STARTING',
+                                             'RUNNING'):
+                resume_instances.append(existing_node)
+
+        for stopped_node in stopped_instances:
+            stopped_node_id = stopped_node['inst_id']
+            instance_action_response = compute_client.instance_action(
+                instance_id=stopped_node_id, action='START')
+
+            starting_inst = instance_action_response.data
+            resume_instances.append({
+                'inst_id': starting_inst.id,
+                'name': starting_inst.display_name,
+                'ad': starting_inst.availability_domain,
+                'compartment': starting_inst.compartment_id,
+                'status': starting_inst.lifecycle_state,
+                'oci_tags': starting_inst.freeform_tags,
+            })
+    # end if config.resume_stopped_nodes
+
+    # Try get head id from the existing instances
+    head_instance_id = _get_head_instance_id(resume_instances)
+    logger.debug(f'Check existing head node: {head_instance_id}')
```

Should we handle the case when head instance id is None? We can raise an error here.

---

Makes sense! LGTM.